### DISCUSSION TITLE
Trying to use conjure for cljs repl with figwheel/cider-piggieback

### DISCUSSION THREAD

**Discussion by esciafardini** [Olical/conjure#370] [General]
Working on a project that utilizes the following:
[cider/piggieback "0.4.2"]
[cider/cider-nrepl "0.28.3"]
[figwheel-sidecar "0.5.20" :exclusions [org.clojure/tools.nrepl]]

Members of the team use fireplace like so:
```vim
:Connect nrepl://localhost:7777
:Piggieback (figwheel-sidecar.repl-api/repl-env)
```

Trying to do something similar in Conjure doesn't work, but maybe I'm doing something wrong...
```vim
:ConjureConnect nrepl://localhost:7777
:ConjurePiggieback (figwheel-sidecar.repl-api/repl-env)
```
I get this hanging output in the REPL buffer:
`piggieback: (figwheel-sidecar.repl-api/repl-env)`

Any ideas for debugging?

**Comment by esciafardini** [comment]
<img width="481" alt="Screen Shot 2022-06-25 at 2 03 51 AM" src="https://user-images.githubusercontent.com/51748017/175760732-bbb90249-139e-4cce-a402-dd1445df76a6.png">

**Comment by Olical** [comment]
Hmm not too sure yet but does this help at all? https://gist.github.com/pyrmont/73d10c74d14f26772fd276c38ee3490d

I haven't worked with figwheel etc in a long time so I'll take a while to delve in and fully understand if that recipe doesn't do the trick. That's linked from the shadow-cljs conjure wiki page but it's kinda hidden https://github.com/Olical/conjure/wiki/Quick-start:-ClojureScript-(shadow-cljs)

**Reply by esciafardini** [reply]
Yeah, I was trying to use that.  Thanks for sharing it.
I'm just going to use emacs for cljs & hope one day our codebase gets off figwheel

**Reply by Olical** [reply]
That's a shame 😢 but I hope it works well for you! I've definitely used Conjure with figwheel before, just a very long time ago and I don't have the time to investigate the issue deeply this week since I have some commitments to make for my Clojurists Together funding.

Just out of interest (even though you'll use another tool) are you using the new figwheel or the original one? I vaguely remember there being a "figwheel next" which was very different?

**Reply by esciafardini** [reply]
`[lein-figwheel "0.5.20"]`
It's okay - I very rarely need to use a REPL for cljs code, so it's only a mild inconvenience.  Thanks :)

**Reply by Samy-33** [reply]
I faced the same issue. I was able to piggieback using `figwheel-sidecar.repl-api/repl-env`
The only change I made was:
```
- :ConjurePiggieback (figwheel-sidecar.repl-api/repl-env :build-id)
+ :ConjurePiggieback (figwheel-sidecar.repl-api/repl-env "build-id")
```
I still don't understand why this is the case. But at least, I've got it working after more than 10 hours of debugging :smiling_face_with_tear:

@esciafardini, if you still work with `lein-figwheel` and Conjure, can you please validate this?

**Reply by esciafardini** [reply]
@Samy-33 my team switched over to shadow-cljs.  

I think the solution I used was jacking in via the port number

:ConjureConnect 7888

Then evaluating this code (which I think I stole from Calva):
```clojure
(do
  (use 'figwheel-sidecar.repl-api)
  (when-not (figwheel-sidecar.repl-api/figwheel-running?)
    (figwheel-sidecar.repl-api/start-figwheel!))
  (figwheel-sidecar.repl-api/cljs-repl))

(js/alert "Hellooooooo!?!?!?")
(.log js/console "check check")
(fig-status)

```
